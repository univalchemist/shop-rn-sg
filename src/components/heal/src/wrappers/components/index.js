import React from 'react';

export {
  Image,
  Text,
  PlainText,
  Box,
  Button,
  Divider,
  Flex,
  SectionHeadingText,
  ErrorPanel,
  ListItemWithRightChevron,
  ScrollView,
  Icon,
  Loader,
  StatusPanel,
} from '@cxa-rn/components';
export {
  withTracking,
  TrackedButton,
  TrackedFloatingActionButton,
  TrackedListItem,
  TrackedCarousel,
  TrackedCarouselWithScrollBar,
} from './Tracking';
