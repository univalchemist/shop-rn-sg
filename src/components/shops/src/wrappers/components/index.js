/* istanbul ignore file */
import React from 'react';
import { useTheme } from '@shops/wrappers/core/hooks';

export {
  Box,
  TouchableContainer,
  Image,
  Card,
  ImageErrorCard,
  ListItem,
  Icon,
  Flex,
  Footer,
  ScrollView,
  Divider,
  Button,
  Input,
  CustomSelectField,
  Carousel,
  TextSkeletonPlaceholder,
} from '@cxa-rn/components';

import {
  Text as TextDefault,
  SectionHeadingText as SectionHeadingTextDefault,
  PlainText as PlainTextDefault,
  SecondaryText as SecondaryTextDefault,
  StackBackButton as StackBackButtonDefault,
} from '@cxa-rn/components';
import { TouchableOpacity } from 'react-native';

export { TrackedButton, ModalBackButton } from '@wrappers/components';
export { InputField } from '@wrappers/components/FinalForm';
export { CustomMultiselectCheckBox } from '@wrappers/components/form';

export const Text = props => {
  const theme = useTheme();

  return <TextDefault {...props} color={props.color || theme.colors.text} />;
};

export const StackBackButton = ({
  onPress,
  icName = 'keyboard-arrow-left',
  ...props
}) => {
  return (
    <TouchableOpacity activeOpacity={1} onPress={onPress} {...props}>
      <StackBackButtonDefault icName={icName} />
    </TouchableOpacity>
  );
};

export const SectionHeadingText = props => {
  const theme = useTheme();

  return (
    <SectionHeadingTextDefault
      {...props}
      color={props.color || theme.colors.text}
    />
  );
};

export const PlainText = props => {
  const theme = useTheme();

  return (
    <PlainTextDefault {...props} color={props.color || theme.colors.text} />
  );
};

export const SecondaryText = props => {
  const theme = useTheme();

  return (
    <SecondaryTextDefault
      {...props}
      color={props.color || theme.colors.label}
    />
  );
};

export { default as TextArea } from './TextArea';
export { default as TextAreaField } from './TextAreaField';
export { default as CheckBox } from './CheckBox';
