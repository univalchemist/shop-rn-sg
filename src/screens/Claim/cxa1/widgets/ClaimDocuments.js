import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import {
  Box,
  SectionHeadingText,
  DocumentUploader,
} from '@wrappers/components';
import { CLAIM_DETAILS_DOCUMENT_VIEWER_MODAL } from '@routes';

const ClaimDocuments = ({ sections, navigation }) => {
  const renderSection = (section, index) => {
    const { items, field, documentType } = section;
    return (
      <Box key={`section-${index}`}>
        {section.title ? (
          <Box py={3}>
            <SectionHeadingText>{section.title}</SectionHeadingText>
          </Box>
        ) : null}

        <DocumentUploader
          size={items.length}
          data={items.map(file => ({
            ...file,
            type: file.contentType,
            uri: file.uri || '',
            secure: true,
          }))}
          onView={idx => {
            navigation.navigate(CLAIM_DETAILS_DOCUMENT_VIEWER_MODAL, {
              ...items[idx],
              secure: true,
            });
          }}
          accessibilityDocumentType={documentType}
          accessibilityField={field}
          displayDot={false}
        />
      </Box>
    );
  };

  return (
    <Box style={styles.container}>
      {sections.map((section, index) => renderSection(section, index))}
    </Box>
  );
};

ClaimDocuments.defaultProps = {
  sections: [],
};

ClaimDocuments.propTypes = {
  sections: PropTypes.arrayOf(PropTypes.shape({})),
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
  },
});

export default ClaimDocuments;
