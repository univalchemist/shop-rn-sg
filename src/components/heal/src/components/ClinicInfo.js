import React from 'react';
import { StyleSheet } from 'react-native';

import { Box, PlainText, Image } from '@heal/src/wrappers/components';
import { location, iconDoctor, appointmentCalendar } from '@heal/images';
import theme from '@theme';
import moment from 'moment';
import { useIntl } from '@heal/src/wrappers/core/hooks';

const styles = StyleSheet.create({
  text: {
    fontSize: 14,
    lineHeight: 20,
    letterSpacing: 0.25,
    color: theme.colors.gray[2],
  },
});

const ClinicInfo = ({
  doctorName,
  speciality,
  address,
  showSpeciality,
  style,
  date,
}) => {
  const intl = useIntl();
  const dateFormat = 'DD MMM YYYY';

  return (
    <Box style={style}>
      <Box flexDirection={'row'} pt={20} mb={25}>
        <Image
          source={appointmentCalendar}
          style={{ tintColor: theme.colors.gray[0] }}
        />
        <Box>
          <PlainText style={styles.text} ml={18}>
            {moment(date).format(dateFormat)}
          </PlainText>
        </Box>
      </Box>
      <Box flexDirection="row" mr="auto" alignItems="center">
        <Image source={iconDoctor} />
        <Box ml={18}>
          <PlainText style={styles.text}>
            {intl.formatMessage({ id: 'heal.appointment.doctorPrefix' }) +
              ' ' +
              doctorName}
          </PlainText>
          {showSpeciality ? (
            <PlainText style={styles.text}>{speciality}r</PlainText>
          ) : null}
        </Box>
      </Box>
      <Box flexDirection="row" mt={24} mr="auto" alignItems="center">
        <Image source={location} />
        <PlainText ml={18} style={styles.text}>
          {address}
        </PlainText>
      </Box>
    </Box>
  );
};

export default ClinicInfo;
