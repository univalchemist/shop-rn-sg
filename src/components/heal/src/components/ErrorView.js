import React from 'react';
import { useIntl, useTheme } from '@heal/src/wrappers/core/hooks';
import { errorPanel } from '@heal/images';
import { Button, Image, StatusPanel } from '@heal/src/wrappers/components';

const ErrorView = ({
  image,
  heading,
  description,
  showButton,
  buttonTitle,
  onPress,
  style = {},
}) => {
  const intl = useIntl();
  const messageImage = image ?? errorPanel;
  const messageHeading =
    heading ?? intl.formatMessage({ id: 'errorPanel.title' });
  const messageDescription =
    description ?? intl.formatMessage({ id: 'errorPanel.message' });

  return (
    <StatusPanel
      style={style}
      image={<Image source={messageImage} />}
      heading={messageHeading}
      description={messageDescription}
      actions={
        showButton && (
          <Button
            primary
            title={
              buttonTitle ??
              intl.formatMessage({
                id: 'goBack',
              })
            }
            onPress={onPress}
          />
        )
      }
    />
  );
};

export default ErrorView;
