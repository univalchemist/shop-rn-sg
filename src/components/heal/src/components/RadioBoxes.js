import React, { useEffect, useState } from 'react';
import { StyleSheet } from 'react-native';
import { Box, Flex, Text } from '@heal/src/wrappers/components';
import { CheckBox } from 'react-native-elements';

import theme from '@theme';

/*
  data: format [{key, value, forceSelected, disable, disableMessage}]
  value: selected key
  onValueChange: return key
*/

const styles = StyleSheet.create({
  radioBox: {
    marginTop: 'auto',
  },
  text: {
    marginTop: 'auto',
  },
});

const RadioBoxes = ({ data, style, value, onValueChange }) => {
  const [selectedValue, setSelectedValue] = useState(null);

  useEffect(() => {
    setSelectedValue(value);
  }, [value]);

  return (
    <Box style={style}>
      {data.map(
        ({ key, value, forceSelected, disable, disableMessage }, index) => {
          return (
            <Box ml={-20} alignItems="center" flexDirection="row">
              <CheckBox
                containerStyle={disable ? styles.radioBox : null}
                checkedIcon="dot-circle-o"
                uncheckedIcon="circle-o"
                checked={forceSelected ? true : key === selectedValue}
                checkedColor={
                  disable
                    ? theme.heal.radioButton.disable
                    : theme.heal.radioButton.checked
                }
                onPress={() => {
                  if (disable) return;
                  setSelectedValue(key);
                  if (onValueChange) onValueChange({ key, index });
                }}
              />
              <Box paddingRight={10} style={disable ? styles.text : null}>
                <Text
                  fontSize={16}
                  color={
                    disable
                      ? theme.heal.radioButton.disable
                      : theme.heal.colors.gray[3]
                  }
                >
                  {value}
                </Text>
                {disable ? (
                  <Text
                    color={theme.heal.colors.gray[0]}
                    fontSize={14}
                    lineHeight={20}
                    letterSpacing={0.25}
                  >
                    {disableMessage}
                  </Text>
                ) : null}
              </Box>
            </Box>
          );
        },
      )}
    </Box>
  );
};

export default RadioBoxes;
