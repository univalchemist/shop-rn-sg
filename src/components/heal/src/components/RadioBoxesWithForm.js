import React, { useEffect, useState } from 'react';
import { StyleSheet } from 'react-native';
import { Box, Flex, Text } from '@heal/src/wrappers/components';
import { CheckBox } from 'react-native-elements';
import { Field } from 'redux-form';

import theme from '@theme';

/*
  data: format [{key, value, forceSelected, disable, disableMessage}]
  value: selected key
  onValueChange: return key
*/

const styles = StyleSheet.create({
  radioBox: {
    marginTop: 'auto',
  },
  text: {
    marginTop: 'auto',
  },
});

const CustomRadioBoxes = props => {
  const {
    input: { value, onChange, name, ...restInput },
    data,
    style,
    ...restProps
  } = props;

  const renderCheckBox = ({
    optionKey,
    optionValue,
    forceSelected,
    disable,
    disableMessage,
    index,
  }) => {
    return (
      <Box ml={-20} alignItems="center" flexDirection="row">
        <CheckBox
          name={name}
          containerStyle={disable ? styles.radioBox : null}
          checkedIcon="dot-circle-o"
          uncheckedIcon="circle-o"
          checked={forceSelected ? true : optionKey === value}
          checkedColor={
            disable
              ? theme.heal.radioButton.disable
              : theme.heal.radioButton.checked
          }
          onPress={() => {
            if (disable) return;
            onChange(optionKey);
          }}
          {...restInput}
          {...restProps}
        />
        <Box paddingRight={30} style={disable ? styles.text : null}>
          <Text
            fontSize={16}
            color={
              disable
                ? theme.heal.radioButton.disable
                : theme.heal.colors.gray[3]
            }
          >
            {optionValue}
          </Text>
          {disable ? (
            <Text
              color={theme.heal.colors.gray[0]}
              fontSize={14}
              lineHeight={20}
              letterSpacing={0.25}
            >
              {disableMessage}
            </Text>
          ) : null}
        </Box>
      </Box>
    );
  };

  return (
    <Box style={style}>
      {data.map(
        ({ key, value, forceSelected, disable, disableMessage }, index) => {
          return renderCheckBox({
            optionKey: key,
            optionValue: value,
            forceSelected,
            disable,
            disableMessage,
          });
        },
      )}
    </Box>
  );
};

const RadioBoxesWithForm = props => {
  return <Field {...props} component={CustomRadioBoxes} />;
};

export default RadioBoxesWithForm;
