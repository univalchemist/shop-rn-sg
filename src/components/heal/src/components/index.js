import { StackBackButton, healNavigationOptions } from './NavigationButtons';
import NonTouchableSearchBar from './NonTouchableSearchBar';
import DoctorCard from './DoctorCard';
import ClinicInfo from './ClinicInfo';
export { default as ErrorView } from './ErrorView';
export { default as SpinningLoader } from './SpinningLoader';
export { default as RadioBoxes } from './RadioBoxes';
export { default as RadioBoxesWithForm } from './RadioBoxesWithForm';
export {
  DoctorCard,
  StackBackButton,
  healNavigationOptions,
  NonTouchableSearchBar,
  ClinicInfo,
};
