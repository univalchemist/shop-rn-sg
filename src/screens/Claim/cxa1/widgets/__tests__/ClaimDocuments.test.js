import React from 'react';
import { renderForTest } from '@testUtils';
import {
  act,
  fireEvent,
  flushMicrotasksQueue,
} from 'react-native-testing-library';
import navigation from '@testUtils/__mocks__/navigation';

import { UploadBox } from '@wrappers/components';
import ClaimDocuments from '../ClaimDocuments';

describe('ClaimDocuments', () => {
  const documents = [
    {
      title: 'Section title 1',
      items: [
        {
          id: 'receipt-1',
          contentType: 'image',
          fileName: 'image2.png',
          uri: 'mock-uri',
        },
        {
          id: 'receipt-2',
          contentType: 'image',
          fileName: 'image2.png',
          uri: 'mock-uri',
        },
      ],
      field: 'section1',
      isDocumentSection: true,
      documentType: 'document',
    },
    {
      title: '',
      items: [
        {
          id: 'referral-1',
          contentType: 'image',
          fileName: 'image2.png',
          uri: 'mock-uri',
        },
        {
          id: 'referral-2',
          contentType: 'image',
          fileName: 'image2.png',
          uri: 'mock-uri',
        },
      ],
      field: 'section2',
      isDocumentSection: true,
      documentType: 'document',
    },
  ];

  it('should render claim documents', async () => {
    const component = renderForTest(
      <ClaimDocuments sections={documents} navigation={navigation} />,
    );
    expect(component.toJSON()).toMatchSnapshot();

    // const documents = component.queryAllByType(UploadBox);

    await flushMicrotasksQueue();
    act(() => {
      fireEvent(component.getAllByType(UploadBox)[0], 'view');
    });
    expect(navigation.navigate).toHaveBeenCalled();
  });
});
