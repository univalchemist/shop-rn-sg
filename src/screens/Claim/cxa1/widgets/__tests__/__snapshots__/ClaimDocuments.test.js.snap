// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimDocuments should render claim documents 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {},
                            Object {
                              "width": "100%",
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <View
                            py={3}
                            style={
                              Array [
                                Object {
                                  "paddingBottom": 16,
                                  "paddingTop": 16,
                                },
                              ]
                            }
                          >
                            <Text
                              color="gray.0"
                              fontFamily="default"
                              fontSize={16}
                              fontWeight="bold"
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#212121",
                                    "fontFamily": "Arial",
                                    "fontSize": 16,
                                    "fontWeight": "700",
                                    "lineHeight": 22,
                                  },
                                ]
                              }
                            >
                              Section title 1
                            </Text>
                          </View>
                          <RCTScrollView
                            horizontal={true}
                            onContentSizeChange={[Function]}
                            onLayout={[Function]}
                            onScroll={[Function]}
                            scrollEventThrottle={0.0001}
                            showsHorizontalScrollIndicator={false}
                          >
                            <View>
                              <View
                                flexDirection="row"
                                flexWrap="nowrap"
                                pr={32}
                                style={
                                  Array [
                                    Object {
                                      "display": "flex",
                                      "flexDirection": "row",
                                      "flexWrap": "nowrap",
                                      "paddingRight": 32,
                                    },
                                  ]
                                }
                              >
                                <View
                                  ml={32}
                                  my={8}
                                  style={
                                    Array [
                                      Object {
                                        "marginBottom": 8,
                                        "marginLeft": 32,
                                        "marginTop": 8,
                                      },
                                    ]
                                  }
                                >
                                  <View>
                                    <View
                                      backgroundColor="white"
                                      height={176}
                                      style={
                                        Array [
                                          Object {
                                            "backgroundColor": "#ffffff",
                                            "borderRadius": 4,
                                            "height": 176,
                                            "shadowColor": "#000",
                                            "shadowOffset": Object {
                                              "height": 2,
                                              "width": 0,
                                            },
                                            "shadowOpacity": 0.05,
                                            "shadowRadius": 3,
                                            "width": 128,
                                          },
                                        ]
                                      }
                                      width={128}
                                    >
                                      <View
                                        alignItem="center"
                                        borderRadius="4px"
                                        height={176}
                                        justifyContent="center"
                                        style={
                                          Array [
                                            Object {
                                              "borderRadius": 4,
                                              "height": 176,
                                              "justifyContent": "center",
                                              "width": 128,
                                            },
                                          ]
                                        }
                                        width={128}
                                      >
                                        <Image
                                          height={176}
                                          source={
                                            Object {
                                              "headers": Object {
                                                "Authorization": "Bearer undefined",
                                              },
                                              "uri": "mock-uri",
                                            }
                                          }
                                          style={
                                            Array [
                                              Object {
                                                "height": 176,
                                              },
                                              Object {
                                                "borderRadius": 4,
                                              },
                                            ]
                                          }
                                        />
                                      </View>
                                      <View
                                        accessibilityLabel="View document of section1"
                                        accessible={true}
                                        focusable={true}
                                        isTVSelectable={true}
                                        onClick={[Function]}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        style={
                                          Array [
                                            Object {
                                              "borderRadius": 4,
                                            },
                                            Object {
                                              "bottom": 0,
                                              "left": 0,
                                              "opacity": 0.5,
                                              "position": "absolute",
                                              "right": 0,
                                              "top": 0,
                                              "zIndex": 2,
                                            },
                                          ]
                                        }
                                      >
                                        <View
                                          style={null}
                                        />
                                      </View>
                                    </View>
                                  </View>
                                </View>
                                <View
                                  ml={16}
                                  my={8}
                                  style={
                                    Array [
                                      Object {
                                        "marginBottom": 8,
                                        "marginLeft": 16,
                                        "marginTop": 8,
                                      },
                                    ]
                                  }
                                >
                                  <View>
                                    <View
                                      backgroundColor="white"
                                      height={176}
                                      style={
                                        Array [
                                          Object {
                                            "backgroundColor": "#ffffff",
                                            "borderRadius": 4,
                                            "height": 176,
                                            "shadowColor": "#000",
                                            "shadowOffset": Object {
                                              "height": 2,
                                              "width": 0,
                                            },
                                            "shadowOpacity": 0.05,
                                            "shadowRadius": 3,
                                            "width": 128,
                                          },
                                        ]
                                      }
                                      width={128}
                                    >
                                      <View
                                        alignItem="center"
                                        borderRadius="4px"
                                        height={176}
                                        justifyContent="center"
                                        style={
                                          Array [
                                            Object {
                                              "borderRadius": 4,
                                              "height": 176,
                                              "justifyContent": "center",
                                              "width": 128,
                                            },
                                          ]
                                        }
                                        width={128}
                                      >
                                        <Image
                                          height={176}
                                          source={
                                            Object {
                                              "headers": Object {
                                                "Authorization": "Bearer undefined",
                                              },
                                              "uri": "mock-uri",
                                            }
                                          }
                                          style={
                                            Array [
                                              Object {
                                                "height": 176,
                                              },
                                              Object {
                                                "borderRadius": 4,
                                              },
                                            ]
                                          }
                                        />
                                      </View>
                                      <View
                                        accessibilityLabel="View document of section1"
                                        accessible={true}
                                        focusable={true}
                                        isTVSelectable={true}
                                        onClick={[Function]}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        style={
                                          Array [
                                            Object {
                                              "borderRadius": 4,
                                            },
                                            Object {
                                              "bottom": 0,
                                              "left": 0,
                                              "opacity": 0.5,
                                              "position": "absolute",
                                              "right": 0,
                                              "top": 0,
                                              "zIndex": 2,
                                            },
                                          ]
                                        }
                                      >
                                        <View
                                          style={null}
                                        />
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <RCTScrollView
                            horizontal={true}
                            onContentSizeChange={[Function]}
                            onLayout={[Function]}
                            onScroll={[Function]}
                            scrollEventThrottle={0.0001}
                            showsHorizontalScrollIndicator={false}
                          >
                            <View>
                              <View
                                flexDirection="row"
                                flexWrap="nowrap"
                                pr={32}
                                style={
                                  Array [
                                    Object {
                                      "display": "flex",
                                      "flexDirection": "row",
                                      "flexWrap": "nowrap",
                                      "paddingRight": 32,
                                    },
                                  ]
                                }
                              >
                                <View
                                  ml={32}
                                  my={8}
                                  style={
                                    Array [
                                      Object {
                                        "marginBottom": 8,
                                        "marginLeft": 32,
                                        "marginTop": 8,
                                      },
                                    ]
                                  }
                                >
                                  <View>
                                    <View
                                      backgroundColor="white"
                                      height={176}
                                      style={
                                        Array [
                                          Object {
                                            "backgroundColor": "#ffffff",
                                            "borderRadius": 4,
                                            "height": 176,
                                            "shadowColor": "#000",
                                            "shadowOffset": Object {
                                              "height": 2,
                                              "width": 0,
                                            },
                                            "shadowOpacity": 0.05,
                                            "shadowRadius": 3,
                                            "width": 128,
                                          },
                                        ]
                                      }
                                      width={128}
                                    >
                                      <View
                                        alignItem="center"
                                        borderRadius="4px"
                                        height={176}
                                        justifyContent="center"
                                        style={
                                          Array [
                                            Object {
                                              "borderRadius": 4,
                                              "height": 176,
                                              "justifyContent": "center",
                                              "width": 128,
                                            },
                                          ]
                                        }
                                        width={128}
                                      >
                                        <Image
                                          height={176}
                                          source={
                                            Object {
                                              "headers": Object {
                                                "Authorization": "Bearer undefined",
                                              },
                                              "uri": "mock-uri",
                                            }
                                          }
                                          style={
                                            Array [
                                              Object {
                                                "height": 176,
                                              },
                                              Object {
                                                "borderRadius": 4,
                                              },
                                            ]
                                          }
                                        />
                                      </View>
                                      <View
                                        accessibilityLabel="View document of section2"
                                        accessible={true}
                                        focusable={true}
                                        isTVSelectable={true}
                                        onClick={[Function]}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        style={
                                          Array [
                                            Object {
                                              "borderRadius": 4,
                                            },
                                            Object {
                                              "bottom": 0,
                                              "left": 0,
                                              "opacity": 0.5,
                                              "position": "absolute",
                                              "right": 0,
                                              "top": 0,
                                              "zIndex": 2,
                                            },
                                          ]
                                        }
                                      >
                                        <View
                                          style={null}
                                        />
                                      </View>
                                    </View>
                                  </View>
                                </View>
                                <View
                                  ml={16}
                                  my={8}
                                  style={
                                    Array [
                                      Object {
                                        "marginBottom": 8,
                                        "marginLeft": 16,
                                        "marginTop": 8,
                                      },
                                    ]
                                  }
                                >
                                  <View>
                                    <View
                                      backgroundColor="white"
                                      height={176}
                                      style={
                                        Array [
                                          Object {
                                            "backgroundColor": "#ffffff",
                                            "borderRadius": 4,
                                            "height": 176,
                                            "shadowColor": "#000",
                                            "shadowOffset": Object {
                                              "height": 2,
                                              "width": 0,
                                            },
                                            "shadowOpacity": 0.05,
                                            "shadowRadius": 3,
                                            "width": 128,
                                          },
                                        ]
                                      }
                                      width={128}
                                    >
                                      <View
                                        alignItem="center"
                                        borderRadius="4px"
                                        height={176}
                                        justifyContent="center"
                                        style={
                                          Array [
                                            Object {
                                              "borderRadius": 4,
                                              "height": 176,
                                              "justifyContent": "center",
                                              "width": 128,
                                            },
                                          ]
                                        }
                                        width={128}
                                      >
                                        <Image
                                          height={176}
                                          source={
                                            Object {
                                              "headers": Object {
                                                "Authorization": "Bearer undefined",
                                              },
                                              "uri": "mock-uri",
                                            }
                                          }
                                          style={
                                            Array [
                                              Object {
                                                "height": 176,
                                              },
                                              Object {
                                                "borderRadius": 4,
                                              },
                                            ]
                                          }
                                        />
                                      </View>
                                      <View
                                        accessibilityLabel="View document of section2"
                                        accessible={true}
                                        focusable={true}
                                        isTVSelectable={true}
                                        onClick={[Function]}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        style={
                                          Array [
                                            Object {
                                              "borderRadius": 4,
                                            },
                                            Object {
                                              "bottom": 0,
                                              "left": 0,
                                              "opacity": 0.5,
                                              "position": "absolute",
                                              "right": 0,
                                              "top": 0,
                                              "zIndex": 2,
                                            },
                                          ]
                                        }
                                      >
                                        <View
                                          style={null}
                                        />
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
